NuGet/ChustaSoft.Tools.Authorization/Services/UserService.cs

```diff
@@ -256,6 +297,45 @@ private async Task TryPerformFakeEmailActions(TUser user)
 
         #endregion
 
+        #region Private methods
+
+        private void manageIdentityResult(IdentityResult result, string actionName)
+        {
+            if (result != IdentityResult.Success)
+            {
+                string errorMessage = result.Errors != null && result.Errors.Count() > 0 ? result.Errors.First().Description : string.Empty;
+                throw new AuthenticationException($"Unable to {actionName} user. {errorMessage}");
+            }
+        }
+
+        private string NormalizeUsername(ExternalLoginInfo loginInfo)
```

Comment:
Create Extension method for ExternalLoginInfo  doing that logic.

This way is easy to test and simplify responsibilities and logic inside the Key Services

Reply:
Logic moved to UserExtensions